internal/auditlog/https_writer.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package auditlog
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/corazawaf/coraza/v3/experimental/plugins/plugintypes"
+)
+
+// httpsWriter is used to store logs in a single file
+type httpsWriter struct {
+	io.Closer
+	formatter plugintypes.AuditLogFormatter
+	url       string
+	client    *http.Client
+}
+
+func (h *httpsWriter) Init(c plugintypes.AuditLogConfig) error {
+	h.Closer = NoopCloser
+	h.formatter = c.Formatter
+	h.url = c.Target
+	// now we validate h.url is a valid url
+	// Although the writer type is HTTPS, we allow HTTP as well
+	_, err := url.Parse(h.url)
+	if err != nil {
+		return err
+	}
+	h.client = &http.Client{
+		Timeout: time.Duration(1 * time.Second),
+	}
+	return nil
+}
+
+func (h *httpsWriter) Write(al plugintypes.AuditLog) error {
+	if h.formatter == nil {
```

Comment:
Shall we check if formatter is nil here? Can't we do it on init?

Reply:
I will take a look, because I remember there was a reason for this, there was no default formatter, causing nil pointer panic